### PR TITLE
[FrameworkBundle] Move web configuration to PHP

### PR DESCRIPTION
**Pull request by ck-developer**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Ref #37186
| License       | MIT

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+use Symfony\Component\HttpKernel\EventListener\ResponseListener;
+use Symfony\Component\HttpKernel\EventListener\StreamedResponseListener;
+use Symfony\Component\HttpKernel\EventListener\ValidateRequestListener;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->defaults()->private()
+
+        ->set('controller_resolver', ControllerResolver::class)
+        ->args([
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php:36]: Please indent each line after `->set()` with extra indent, check the reference CS here https://github.com/symfony/symfony/pull/36778/files#diff-7e8f7b3f9e57b4e008c70ba7db12d142R20

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+use Symfony\Component\HttpKernel\EventListener\DisallowRobotsIndexingListener;
+use Symfony\Component\HttpKernel\EventListener\ErrorListener;
+use Symfony\Component\HttpKernel\EventListener\LocaleListener;
+use Symfony\Component\HttpKernel\EventListener\ResponseListener;
+use Symfony\Component\HttpKernel\EventListener\StreamedResponseListener;
+use Symfony\Component\HttpKernel\EventListener\ValidateRequestListener;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->defaults()->private()
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php:33]: We can get rid of this line as it's already the default (same for web_link.php)

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+            service('service_container'),
+            service('logger')->ignoreOnInvalid(),
+        ])
+        ->tag('monolog.logger', ['channel' => 'request'])
+
+        ->set('argument_metadata_factory', ArgumentMetadataFactory::class)
+
+        ->set('argument_resolver', ArgumentResolver::class)
+        ->args([
+            service('argument_metadata_factory'),
```
Comment (yceruto) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php:46]: let's add `abstract_arg()` for empty argument definition in XML.

In this case, can be something like this `abstract_arg('argument value resolvers')`

Ref: https://github.com/symfony/symfony/pull/36778/files#diff-c9620897a07dd9bcada6305d3623c687R151

**Review by yceruto — CHANGES_REQUESTED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php (hunk 2 above)
Comment (ck-developer) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/web.php]: Hi @yceruto , thank you for your review. it's all fixed

**Review by yceruto — APPROVED** [review]
LGTM, thanks!

**Review by javiereguiluz — APPROVED** [review]
Looks good! Thanks!

The only remaining question is if we should put all arguments of `args([ ... ])` in a single line when there are only a few of them.

**Comment by Tobion** [comment]
> The only remaining question is if we should put all arguments of args([ ... ]) in a single line when there are only a few of them.

Using multiple lines makes merging easier and it reduces conflicts when you add a new argument.

**Review by Tobion — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml [hunk 4]
```diff
@@ -1,86 +0,0 @@
...
-            <tag name="controller.argument_value_resolver" priority="50" />
-        </service>
-
-        <service id="argument_resolver.session" class="Symfony\Component\HttpKernel\Controller\ArgumentResolver\SessionValueResolver">
-            <tag name="controller.argument_value_resolver" priority="50" />
-        </service>
-
-        <service id="argument_resolver.service" class="Symfony\Component\HttpKernel\Controller\ArgumentResolver\ServiceValueResolver">
-            <tag name="controller.argument_value_resolver" priority="-50" />
-            <argument />
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml:37]: this is missing an abstract_arg equivalent which makes the tests fail

> Cannot replace arguments if none have been configured yet.

Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/web.xml:37]: Added a commit to fix this

**Comment by Tobion** [comment]
Thank you @ck-developer.

**Pull request merged**